packages/app/src/runner/unifiedRunner.ts

```diff
@@ -25,26 +42,47 @@ export function useUnifiedRunner () {
       const specStore = useSpecStore()
       const route = useRoute()
       const selectorPlaygroundStore = useSelectorPlaygroundStore()
+      const runId = computed(() => route.query.runId as string)
+      const query = useQuery({ query: TestsForRunDocument, pause: true, requestPolicy: 'network-only', variables: { runId: runId as any } })
 
-      watchEffect(() => {
+      watchEffect(async () => {
         const queryFile = getPathForPlatform(route.query.file as string)
 
-        if (!queryFile) {
+        if (!queryFile || isEqual(route.query, prevQuery)) {
           // no file param, we are not showing a file
           // so no action needed when specs list updates
           return
         }
 
+        prevQuery = route.query
+
         if (queryFile === RUN_ALL_SPECS_KEY) {
           return specStore.setActiveSpec(RUN_ALL_SPECS)
         }
 
-        const activeSpecInSpecsList = specs.value.find((x) => x.relative === queryFile)
+        let activeSpecInSpecsList: SpecWithFilter | undefined = specs.value.find((x) => x.relative === queryFile)
+
+        if (!activeSpecInSpecsList) {
+          return specStore.setActiveSpec(null)
+        }
+
+        if (route.query.runId) {
+          const queryRes = await query.executeQuery({ runId: route.query.runId })
+          const testResults = queryRes.data.value?.currentProject?.testsForRun || []
+
+          const failedTests = (queryRes.data.value?.currentProject?.testsForRun || []).reduce<string[]>((acc, test) => {
+            if (test.status === 'FAILED') acc.push(test.titlePath)
+
+            return acc
+          }, [])
```

Comment:
Use filter and map to make it more clear what this is doing
```suggestion
          const failedTests = (testResults)
            .filter((test) => test.status === 'FAILED')
            .map((test) => test.titlePath)
```

Reply:
I yo yo between doing the map + filter and the reduce. If I did it the suggested way I'd most likely get a comment to convert it to a reduce for performance benefits which is where I tend to lean nowadays.

Reply:
I always use reduce over map+filter for this reason. I could go either way here, I don't really mind, but I like to err on the side of reducing un-necessary loops.